### PR TITLE
RISCV: Added SerialServer and TimeServer support for hifive platform

### PR DESCRIPTION
**Pull request by Jesse-Millwood**
These two patches are very close to the arm_common implementations, as they are copied from there and modified accordingly.

### CODE REVIEW

File: components/SerialServer/include/plat/hifive/plat/serial.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
```
Comment (kent-mcleod) [inline components/SerialServer/include/plat/hifive/plat/serial.h]: Are you happy to instead treat the arm_common files as arm/riscv common and update the CMake scripts accordingly?

Comment (Jesse-Millwood) [reply components/SerialServer/include/plat/hifive/plat/serial.h]: yeah that sounds good

**Comment by Jesse-Millwood** [comment]
Rebased and added the riscv-common plat for SerialServer

**Comment by nomadeel** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
## Test results for commit 27a728cea1c8c6263fe589180891c16e9e9b0b59

### Summary

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - global-components - EXT-dornerworks-master-7 | SUCCESSFUL |
| Code Quality - Style - global-components - EXT-dornerworks-master-7 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Style - global-components - EXT-dornerworks-master-7 | Style check on changed files | 1 of 3 jobs failed |

**Job: gitlint**
Build logs

    02-Jan-2020 10:57:54	Build Code Quality - Style - global-components - EXT-dornerworks-master-7 - gitlint #1 (CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1) started building on agent A local agent 1
    02-Jan-2020 10:57:54	I run on the Bamboo server!
    02-Jan-2020 10:57:54	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR
    02-Jan-2020 10:57:54	Executing build Code Quality - Style - global-components - EXT-dornerworks-master-7 - gitlint #1 (CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1)
    02-Jan-2020 10:57:54	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR/. (location: .)
    02-Jan-2020 10:57:54	Artifact 'code' prepared in 22.10 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR/./code.tar.gz
    02-Jan-2020 10:57:54	Running pre-build action: VCS Version Collector
    02-Jan-2020 10:57:55	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:55	+ tar xf code.tar.gz --recursive-unlink
    02-Jan-2020 10:57:55	Finished task 'Untar code' with result: Success
    02-Jan-2020 10:57:55	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:55	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR
    02-Jan-2020 10:57:55	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    02-Jan-2020 10:57:55	Starting task 'Lint commit messages' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:55	+ cd global-components
    02-Jan-2020 10:57:55	+ gitlint --commits master..27a728cea1c8c6263fe589180891c16e9e9b0b59 --config /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR/seL4_tools/misc/.gitlint
    02-Jan-2020 10:57:59	
    02-Jan-2020 10:57:59	Commit eba47cde93:
    02-Jan-2020 10:57:59	3: B1 Line exceeds max length (77>72): "This enables the Serial Server component for the riscv_common platform. These"
    02-Jan-2020 10:57:59	4: B1 Line exceeds max length (78>72): "files were initially copied from the arm_common platform files. The PlatPrefix"
    02-Jan-2020 10:57:59	5: B1 Line exceeds max length (74>72): "is set based whether or not the KernelArchRiscV variable is set. This also"
    02-Jan-2020 10:57:59	6: B1 Line exceeds max length (74>72): "requires that the device tree for the platform has a valid "chosen" serial"
    02-Jan-2020 10:57:59	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1-ScriptBuildTask-8569209464688367165.sh] was 4 while expected 0
    02-Jan-2020 10:57:59	Finished task 'Lint commit messages' with result: Failed
    02-Jan-2020 10:57:59	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:59	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR
    02-Jan-2020 10:58:00	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    02-Jan-2020 10:58:00	Running post build plugin 'NCover Results Collector'
    02-Jan-2020 10:58:00	Running post build plugin 'Artifact Copier'
    02-Jan-2020 10:58:00	Running post build plugin 'npm Cache Cleanup'
    02-Jan-2020 10:58:00	Running post build plugin 'Clover Results Collector'
    02-Jan-2020 10:58:00	Running post build plugin 'Docker Container Cleanup'
    02-Jan-2020 10:58:00	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR'
    02-Jan-2020 10:58:00	Finalising the build...
    02-Jan-2020 10:58:00	Stopping timer.
    02-Jan-2020 10:58:00	Build CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1 completed.
    02-Jan-2020 10:58:00	Running on server: post build plugin 'NCover Results Collector'
    02-Jan-2020 10:58:00	Running on server: post build plugin 'Build Hanging Detection Configuration'
    02-Jan-2020 10:58:00	Running on server: post build plugin 'Clover Delta Calculator'
    02-Jan-2020 10:58:00	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    02-Jan-2020 10:58:00	All post build plugins have finished
    02-Jan-2020 10:58:00	Generating build results summary...
    02-Jan-2020 10:58:00	Saving build results to disk...
    02-Jan-2020 10:58:00	Logging substituted variables...
    02-Jan-2020 10:58:00	Indexing build results...
    02-Jan-2020 10:58:00	Finished building CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1.


This is the most I can report on right now, sorry!

**Comment by nomadeel** [comment]
Looks like the style checker is unhappy that the first commit's body message has lines that exceed the max length: `3: B1 Line exceeds max length (77>72): "This enables the Serial Server component for the riscv_common platform. These"` and so on.

**Comment by ssrg-bamboo** [comment]
## Test results for commit 27a728cea1c8c6263fe589180891c16e9e9b0b59

### Summary

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - global-components - EXT-dornerworks-master-7 | SUCCESSFUL |
| Code Quality - Style - global-components - EXT-dornerworks-master-7 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Style - global-components - EXT-dornerworks-master-7 | Style check on changed files | 1 of 3 jobs failed |

**Job: gitlint**
Build logs

    02-Jan-2020 10:57:54	Build Code Quality - Style - global-components - EXT-dornerworks-master-7 - gitlint #1 (CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1) started building on agent A local agent 1
    02-Jan-2020 10:57:54	I run on the Bamboo server!
    02-Jan-2020 10:57:54	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR
    02-Jan-2020 10:57:54	Executing build Code Quality - Style - global-components - EXT-dornerworks-master-7 - gitlint #1 (CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1)
    02-Jan-2020 10:57:54	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR/. (location: .)
    02-Jan-2020 10:57:54	Artifact 'code' prepared in 22.10 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR/./code.tar.gz
    02-Jan-2020 10:57:54	Running pre-build action: VCS Version Collector
    02-Jan-2020 10:57:55	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:55	+ tar xf code.tar.gz --recursive-unlink
    02-Jan-2020 10:57:55	Finished task 'Untar code' with result: Success
    02-Jan-2020 10:57:55	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:55	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR
    02-Jan-2020 10:57:55	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    02-Jan-2020 10:57:55	Starting task 'Lint commit messages' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:55	+ cd global-components
    02-Jan-2020 10:57:55	+ gitlint --commits master..27a728cea1c8c6263fe589180891c16e9e9b0b59 --config /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR/seL4_tools/misc/.gitlint
    02-Jan-2020 10:57:59	
    02-Jan-2020 10:57:59	Commit eba47cde93:
    02-Jan-2020 10:57:59	3: B1 Line exceeds max length (77>72): "This enables the Serial Server component for the riscv_common platform. These"
    02-Jan-2020 10:57:59	4: B1 Line exceeds max length (78>72): "files were initially copied from the arm_common platform files. The PlatPrefix"
    02-Jan-2020 10:57:59	5: B1 Line exceeds max length (74>72): "is set based whether or not the KernelArchRiscV variable is set. This also"
    02-Jan-2020 10:57:59	6: B1 Line exceeds max length (74>72): "requires that the device tree for the platform has a valid "chosen" serial"
    02-Jan-2020 10:57:59	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1-ScriptBuildTask-8569209464688367165.sh] was 4 while expected 0
    02-Jan-2020 10:57:59	Finished task 'Lint commit messages' with result: Failed
    02-Jan-2020 10:57:59	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    02-Jan-2020 10:57:59	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR
    02-Jan-2020 10:58:00	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    02-Jan-2020 10:58:00	Running post build plugin 'NCover Results Collector'
    02-Jan-2020 10:58:00	Running post build plugin 'Artifact Copier'
    02-Jan-2020 10:58:00	Running post build plugin 'npm Cache Cleanup'
    02-Jan-2020 10:58:00	Running post build plugin 'Clover Results Collector'
    02-Jan-2020 10:58:00	Running post build plugin 'Docker Container Cleanup'
    02-Jan-2020 10:58:00	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR'
    02-Jan-2020 10:58:00	Finalising the build...
    02-Jan-2020 10:58:00	Stopping timer.
    02-Jan-2020 10:58:00	Build CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1 completed.
    02-Jan-2020 10:58:00	Running on server: post build plugin 'NCover Results Collector'
    02-Jan-2020 10:58:00	Running on server: post build plugin 'Build Hanging Detection Configuration'
    02-Jan-2020 10:58:00	Running on server: post build plugin 'Clover Delta Calculator'
    02-Jan-2020 10:58:00	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    02-Jan-2020 10:58:00	All post build plugins have finished
    02-Jan-2020 10:58:00	Generating build results summary...
    02-Jan-2020 10:58:00	Saving build results to disk...
    02-Jan-2020 10:58:00	Logging substituted variables...
    02-Jan-2020 10:58:00	Indexing build results...
    02-Jan-2020 10:58:00	Finished building CQ-GLOBALCOMPONENTSSTYLE43-GLOBALCOMPONENTSGITLINTPR-1.


This is the most I can report on right now, sorry!

**Comment by axel-h** [comment]
@Jesse-Millwood: do you plan to address the bamboo remarks?

**Comment by Jesse-Millwood** [comment]
> @Jesse-Millwood: do you plan to address the bamboo remarks?

Yeah I'll be addressing this this coming week. Thanks for the ping.

**Comment by Jesse-Millwood** [comment]
@ssrg-bamboo `test`

**Comment by kent-mcleod** [comment]
@ssrg-bamboo `test`

**Comment by lsf37** [comment]
The bot seems to have gotten stuck for some reason. I synced the repo manually now and hopefully that will make it unstuck..

@ssrg-bamboo `test`

**Comment by lsf37** [comment]
Ok, this is getting silly. The previously failing gitlint check now passes fine, so I'll merge this and look at adding the GitHub checks on this repo later today.

**Comment by lsf37** [comment]
After some investigation we get the following: the bot thought (and still thinks) that the PR is closed. I don't think it has to do with the close/re-open further in the past, because the bot wasn't pinged there, and it did work afterwards. It is possible that there was a network glitch when it tried for Jesse, and after having decided the PR is closed, it removes it from its notifications.

**Pull request merged**